src/env.mjs

```diff
@@ -32,6 +32,20 @@ export const env = createEnv({
    */
   client: {
     // NEXT_PUBLIC_CLIENTVAR: z.string().min(1),
+    NEXT_PUBLIC_SITE_URL: z
+      .optional(z.string())
+      .transform((str) => {
+        let url =
+          str ??
+          process.env.NEXT_PUBLIC_VERCEL_URL ?? // Automatically set by Vercel.
+          "http://localhost:3000/";
+        // Make sure to include `https://` when not localhost.
+        url = url.includes("http") ? url : `https://${url}`;
```

Comment:
Suggestion: Would it be better to use `url.startsWith("http")` in this case? Since we will always want to check the start of the string, using `startsWith` might be more appropriate

Reply:
great point! updated at aa17b9a219d9c9b5bef08309a784cdcdc479a0a3